File: app/src/main/java/com/frank/protean/mode/MacbookBuilder.java
package com.frank.protean.mode;

public class MacbookBuilder extends Builder {
    protected Computer computer = new Macbook();

    @Override
    public Builder buildBoard(String board) {
        computer.setmBoard(board);
        return this;
    }

    @Override
    public Builder buildDisplay(String display) {
        computer.setmDisplay(display);
        return this;
    }

    @Override
    public Builder buildOS() {
        computer.setmOS();
        return this;
    }

    @Override
    public Computer create() {
        return computer;
    }
}

File: app/src/main/java/com/frank/protean/mode/builder/MacbookBuilder.kt
package com.frank.protean.mode.builder

class MacbookBuilder : Builder {
    private val macbook: Computer = Macbook()
    override fun buildDisplay(disply: String?): Builder {
        macbook.display = disply
        return this
    }

    override fun buildBoard(board: String?): Builder {
        macbook.board = board
        return this
    }

    override fun buildOs(): Builder {
        macbook.setOs()
        return this
    }

    override fun build(): Computer? {
        return macbook
    }

}